subseas_flex_fcst/maproom.py

```diff
@@ -96,26 +96,16 @@ def display_relevant_control(variable):
     Output("leadTime","value"),
     Input("startDate","value"),
 )
-def targetStartOptions(startDate):
-    leadsValues = list(CONFIG["leads"].values())
-    leadsKeys = list(CONFIG["leads"])
-    startDate = pd.to_datetime(startDate)
-    optionsDict = {}
-    for idx, x in enumerate(leadsValues):
-        endDate = x+CONFIG["target_period_length"]-1 #used to calculate the target end date (targetEnd = lead + period length - 1)
-        targetStart = startDate + timedelta(days=x)
-        targetEnd = startDate + timedelta(days=endDate)
-        if (targetStart).strftime("%Y") == (targetEnd).strftime("%Y"):
-            if (targetStart).strftime("%b") == (targetEnd).strftime("%b"):
-                targetStartStr = targetStart.strftime("%-d")
-            else:
-                targetStartStr = (targetStart).strftime("%-d %b")
-        else:
-            targetStartStr = (targetStart).strftime("%-d %b %Y")
-        targetEndStr = targetEnd.strftime("%-d %b %Y")
-        dateRange = f"{targetStartStr} - {targetEndStr}"
-        optionsDict.update({leadsKeys[idx]:dateRange})
-    return optionsDict, leadsKeys[0]
+def targetStartOptions(start_date):
+    leads_values = list(CONFIG["leads"].values())
+    leads_keys = list(CONFIG["leads"])
+    start_date = pd.to_datetime(start_date)
+    leads_dict = {}
+    for idx, lead in enumerate(leads_keys):
+        target_range = predictions.target_range_format(leads_values[idx],leads_keys[idx],start_date,CONFIG["target_period_length"])
+        leads_dict.update({lead:target_range})
+    return leads_dict, list(CONFIG["leads"])[0]
```

Comment:
now, your callback outputs the exact a value that you explicitely give in your layout, instead of using the result of your CB. Do one or the other. That is to say either:
1/ Don't output leadTime/value
OR
2/ remove the line value=list(CONFIG["leads"])[0]
in the layout's leadTime
I guess I would do 1/ since the default leadTime value is always the same.

Reply:
So this part is a little buggy; if you do not have the leadTime/value as an output of the CB, then the value does not reset to be the first lead time when the use selects a different start date (for example if the user picks the thirs lead time and then selects a new start date, the leadtime value will not reset back to the first lead time). So as far as I can tell, you need that as an output in the callback. 

So I can do the second option of removing the initial value definition in the leadTime layout. And then leave the callback as-is.

Reply:
My point was mostly that you are not using the output of the CB, unless it's overwritten? But anyways, it's very fine with me that the lead doesn't reset to the first one when users change the issue date. All that makes sense to me is that it is the first one on opening of the page.

Reply:
I think that it is overwritten each time. Personally I think it is better for it to reset each time a user chooses a new start date, for it to go back to the first lead time option. I think I would be a bit confused when first using the app if it it did not do so.

Reply:
I see... I think I came to realize that components are overwritten indeed if they have a value at first.

Resetting to the first lead time looks good. I think it was hard to do with uicore so didn't implement so.

So I guess this clarification resolves that.